tests/test_diffusers.py

```diff
@@ -1329,6 +1329,144 @@ def test_stable_diffusion_xl_hpu_graphs(self):
+
+        prompt = "A painting of a squirrel eating a burger"
+
+        # Test num_images_per_prompt=1 (default)
+        images = sd_pipe(prompt, num_inference_steps=2, output_type="np").images
+
+        self.assertEqual(len(images), 1)
+        self.assertEqual(images[0].shape, (1024, 1024, 3))
+
+        # Test num_images_per_prompt=1 (default) for several prompts
+        num_prompts = 3
+        images = sd_pipe([prompt] * num_prompts, num_inference_steps=2, output_type="np").images
+
+        self.assertEqual(len(images), num_prompts)
+        self.assertEqual(images[-1].shape, (1024, 1024, 3))
+
+        # Test num_images_per_prompt for single prompt
+        num_images_per_prompt = 2
+        images = sd_pipe(
+            prompt, num_inference_steps=2, output_type="np", num_images_per_prompt=num_images_per_prompt
+        ).images
+
+        self.assertEqual(len(images), num_images_per_prompt)
+        self.assertEqual(images[-1].shape, (1024, 1024, 3))
+
+        # Test num_images_per_prompt for several prompts
+        num_prompts = 2
+        images = sd_pipe(
+            [prompt] * num_prompts,
+            num_inference_steps=2,
+            output_type="np",
+            num_images_per_prompt=num_images_per_prompt,
+        ).images
+
+        self.assertEqual(len(images), num_prompts * num_images_per_prompt)
+        self.assertEqual(images[-1].shape, (1024, 1024, 3))
+
+    def test_stable_diffusion_xl_optimized_fp8(self):
+
+        kwargs = {"timestep_spacing": "linspace"}
+        scheduler = GaudiEulerDiscreteScheduler.from_pretrained(
+            "stabilityai/stable-diffusion-xl-base-1.0", subfolder="scheduler", **kwargs
+            )
+        scheduler.hpu_opt = True
+        kwargs = {
+            "scheduler": scheduler,
+            "use_habana": True,
+            "use_hpu_graphs": True,
+            "gaudi_config": "Habana/stable-diffusion",
+            "torch_dtype": torch.bfloat16
+        }
+
+        os.environ["PATCH_SDPA"] = "1"
+        # Set QUANT_CONFIG environment variable
+        os.environ["QUANT_CONFIG"] = "./quantization/quant_config.json"
+
+        from optimum.habana.diffusers.pipelines.stable_diffusion_xl.pipeline_stable_diffusion_xl_mlperf import (
+            StableDiffusionXLPipeline_HPU,
+        )
+
+        model_name_or_path = "stabilityai/stable-diffusion-xl-base-1.0"
+
+        sd_pipe = StableDiffusionXLPipeline_HPU.from_pretrained(
+            model_name_or_path,
+            **kwargs,
+        )
+        logger = logging.get_logger(sd_pipe.__module__)
+        logger.setLevel(level=diffusers.logging.INFO)
+        sd_pipe.unet.set_default_attn_processor(sd_pipe.unet)
+        sd_pipe.to(torch.device("hpu"))
+
+        quant_config_path = os.getenv("QUANT_CONFIG")
+
+        original_dir = os.getcwd()
+        config_dir = os.path.dirname(os.path.abspath("../examples/stable-diffusion/quantization"))
+        os.chdir(config_dir)
```

Comment:
Is this absolutely necessary?

Reply:
yes, this is required to avoid file not found error because we are executing inside from tests/, for below measurement file, it is present in examples/stable-diffusion/quantization
https://github.com/huggingface/optimum-habana/blob/main/examples/stable-diffusion/quantization/quant_config.json#L6

in the end I do restore the original directory